detectable/src/main/java/com/blackduck/integration/detectable/detectables/gradle/inspection/parse/GradleReportTransformer.java

```diff
@@ -24,11 +24,127 @@
+                logger.debug("Adding configuration to the graph: {}", configuration.getName());
+                addConfigurationToRootAndSubProjectGraphs(rootGraph, configuration, gradleReport);
+            } else {
+                logger.trace("Excluding unresolved configuration from the graph: {}", configuration.getName());
+            }
+        }
+
+        ExternalId projectId = ExternalId.FACTORY.createMavenExternalId(gradleReport.getProjectGroup(), gradleReport.getProjectName(), gradleReport.getProjectVersionName());
+        if (StringUtils.isNotBlank(gradleReport.getProjectSourcePath())) {
+            CodeLocation rootCodeLocation = new CodeLocation(rootGraph, projectId, new File(gradleReport.getProjectSourcePath()));
+            allCodeLocationsWithinRoot.add(rootCodeLocation);
+        } else {
+            CodeLocation rootCodeLocation = new CodeLocation(rootGraph, projectId);
+            allCodeLocationsWithinRoot.add(rootCodeLocation);
+        }
+
+        return allCodeLocationsWithinRoot;
+    }
+
+
+    private void addConfigurationToRootAndSubProjectGraphs(DependencyGraph rootGraph, GradleConfiguration configuration, GradleReport rootReport) {
+        DependencyHistory history = new DependencyHistory();
+
+        TreeNodeSkipper treeNodeSkipper = new TreeNodeSkipper();
+        for (GradleTreeNode currentNode : configuration.getChildren()) {
+            if (treeNodeSkipper.shouldSkip(currentNode)) {
+                continue;
+            }
+
+            if (currentNode.getNodeType() == GradleTreeNode.NodeType.GAV) {
+                history.clearDependenciesDeeperThan(currentNode.getLevel());
+                Optional<GradleGav> currentNodeGav = currentNode.getGav();
+                if (currentNodeGav.isPresent()) {
+                    addGavToGraph(currentNodeGav.get(), history, rootGraph);
+                }
+                else {
+                    // We know this is a GradleTreeNode.NodeType.GAV
+                    // So if its missing data, something is probably wrong.
+                    logger.debug("Missing expected GAV from known NodeType. {}", currentNode);
+                }
+            }
+            else if (currentNode.getNodeType() == GradleTreeNode.NodeType.PROJECT) {
+                // Let this method skip over the subProject section as usual
+                treeNodeSkipper.skipUntilLineLevel(currentNode.getLevel());
+                // Process subProject section separately
+                processSubprojectAndCreateCodeLocation(currentNode, configuration.getChildren(), rootReport);
+            } else {
+                treeNodeSkipper.skipUntilLineLevel(currentNode.getLevel());
+            }
+        }
+    }
+    private void processSubprojectAndCreateCodeLocation(GradleTreeNode subProjectNode, List<GradleTreeNode> allTreeNodesInCurrentConfiguration, GradleReport rootReport) {
```

Comment:
It looks like there are opportunities to deduplicate code between `addConfigurationToRootAndSubProjectGraphs` and this function as well.

Reply:
WIP -- sonar also complains of too many conditionals, working on refactoring 🫡

Reply:
There is still code duplication between addConfigurationToRootAndSubProjectGraphs(), addConfigurationToGraph() and processSubprojectAndCreateCodeLocation(). But refactoring that (mainly the TreeNodeSkipper.shouldSkip() logic) was more complicated than it is worth. For any maintanence we do to that logic in the future, it would be beneficial to repeat in the 3 contexts and confirm the logic change would not cause regressions